lib/Backend/GlobOpt.cpp

```diff
@@ -2820,7 +2820,9 @@ GlobOpt::OptTagChecks(IR::Instr *instr)
             {
                 if (valueType.CanBeTaggedValue())
                 {
-                    ChangeValueType(nullptr, value, valueType.SetCanBeTaggedValue(false), false);
+                    // We're not adding new information to the value other than changing the value type. Preserve any existing
+                    // information and just change the value type.
+                    ChangeValueType(nullptr, value, valueType.SetCanBeTaggedValue(false), true);
```

Comment:
>ChangeValueType [](start = 20, length = 15)

The fix by itself seems good. There are other usages of this method that may need fixing or at least understanding why they do not 'preserveSubClassInfo'. Also could you pass the last parameter as "true /*preserveSubClassInfo*/" ?